src/jit/compiler.cpp

```diff
@@ -4194,11 +4194,17 @@ void Compiler::compCompile(void** methodCodePtr, ULONG* methodCodeSize, JitFlags
     assert(!fgComputePredsDone);
     if (fgCheapPredsValid)
     {
-        // Remove cheap predecessors before inlining; allowing the cheap predecessor lists to be inserted
-        // with inlined blocks causes problems.
+        // Remove cheap predecessors before inlining and fat call transformation;
+        // allowing the cheap predecessor lists to be inserted causes problems
+        // with splitting existing blocks.
         fgRemovePreds();
     }
 
+    if (IsTargetAbi(CORINFO_CORERT_ABI) && doesMethodHaveFatPointer())
+    {
+        fgTransformFatCalli();
```

Comment:
I don't understand your doubts.

1. It supposed to be part of importer, because it is transformation from IL calli into our IR.
It doesn't have anything with "Massage the trees so that we can generate code out of them".
Are you are agree that we should set the flag in importer?
2. We don't want to transform cfg inside fgImport, it is why this transformation is done as separate phase. 
3. We are trying to make this transformation as close as possible to importer to avoid any collisions with different optimizations. (for example tail calls or struct promotions).

I can move this phase, it will create many new checks in phases between importer and the new place. Do you want it just to avoid setting BBF_IMPORTED on created blocks?